src/main/java/com/github/hokkaydo/eplbot/module/code/CodeCommand.java

```diff
@@ -0,0 +1,174 @@
+                                                     .addActionRow(TextInput.create("language", "Language", TextInputStyle.PARAGRAPH).setPlaceholder("python|rust|java").setRequired(true).build())
+                                                     .addActionRow(TextInput.create("body", "Code", TextInputStyle.PARAGRAPH).setPlaceholder("Code").setRequired(true).build())
+                                                     .build()).queue();
+        } else {
+            context.replyCallbackAction().setContent("Processing since: <t:" + Instant.now().getEpochSecond() + ":R>").setEphemeral(false).queue();
+            context.options()
+                    .get(0)
+                    .getAsAttachment()
+                    .getProxy()
+                    .downloadToFile(new File("%s\\input.txt".formatted(TEMP_DIR)))
+                    .thenAcceptAsync(file -> {
+                        String content;
+                        try {
+                            content = readFromFile(file);
+                        } catch (IOException e) {
+                            content = "";
+                            e.printStackTrace();
+                        }
+                        Runner runner = RUNNER_MAP.get(context.options().get(1).getAsString());
+                        messageLengthCheck(context.channel(), content, runner.run(content, Config.getGuildVariable(guild.getIdLong(), "COMMAND_CODE_TIMELIMIT")),context.options().get(1).getAsString());
+                        file.delete();
+                    })
+                    .exceptionally(t -> {
+                        t.printStackTrace();
+                        return null;
+                    });
+        }
+    }
+    @Override
+    public void onModalInteraction(@NotNull ModalInteractionEvent event) {
+        if(event.getInteraction().getType() != InteractionType.MODAL_SUBMIT || !event.getModalId().contains("Code")) return;
+        Optional<ModalMapping> lang = Optional.ofNullable(event.getInteraction().getValue("language"));
+        Optional<ModalMapping> body = Optional.ofNullable(event.getInteraction().getValue("body"));
+        Guild guild = event.getGuild();
+        if(lang.isEmpty() || body.isEmpty() || guild == null) return;
+
+        Integer runTimeout = Config.getGuildVariable(guild.getIdLong(), "COMMAND_CODE_TIMELIMIT");
+        event.getInteraction().reply("Processing since: <t:" + Instant.now().getEpochSecond() + ":R>").queue();
+
+        String languageOption = Objects.requireNonNull(lang.get().getAsString());
+        String bodyStr = Objects.requireNonNull(body.get().getAsString());
+        Runner runner = RUNNER_MAP.get(languageOption);
+        messageLengthCheck(event.getMessageChannel(),bodyStr, runner.run(bodyStr, runTimeout),languageOption);
+    }
+    private void messageLengthCheck(MessageChannel textChannel, String bodyStr, String result,String codeName){
```

Comment:
The name of this function suggests we just check the length of the message but there is more. Please split this method into short parts :
- a real `boolean messageLengthCheck(...)` that doesn't use a hacky IllegalState exception to check if the message is too long but instead uses documentation to see "IllegalArgumentException – If the content is null or longer than 2000 characters" and checks if the content is longer than 2000 chars.
- a method to send the message as text (optionally)
- a method to send the message as file